cpp/src/strings/json/json_path.cu

```diff
@@ -327,15 +290,62 @@ class json_state : private parser {
     return parse_result::ERROR;
   }
 
+  // a name means:
+  // - a string followed by a :
+  // - no string
+  CUDA_HOST_DEVICE_CALLABLE parse_result parse_name(string_view& name, bool can_be_empty)
+  {
+    char const quote = options.get_allow_single_quotes() ? 0 : '\"';
+
+    if (parse_string(name, can_be_empty, quote) == parse_result::ERROR) {
+      return parse_result::ERROR;
+    }
+
+    // if we got a real string, the next char must be a :
+    if (name.size_bytes() > 0) {
+      if (!parse_whitespace()) { return parse_result::ERROR; }
+      if (*pos == ':') {
+        pos++;
+        return parse_result::SUCCESS;
+      }
+    }
+    return parse_result::EMPTY;
+  }
+
  private:
+  // numbers, true, false, null.
+  // this function is not particularly strong. badly formed values will get
+  // consumed without throwing any errors
```

Comment:
Make this into a doxygen style comment that also includes a description for the `val` parameter.